### PR TITLE
rpiboot: Add build date and version

### PR DESCRIPTION
**Pull request by timg236**
Update rpiboot to emit the build-date and version in order so that this visible in bug reports.

### CODE REVIEW

**Comment by timg236** [comment]
@peterharperuk @pelwell @XECDesign @dp111

Add version info, old versions of this package won't work with CM4S so add something to capture the version in bug reports. We should update the APT package after this has merged.

**Comment by XECDesign** [comment]
Assuming it's built from a git tree can be a problem for apt packages, which don't ship the .git directory in their source archive.

**Comment by timg236** [comment]
That's a pain. It means that you can never know the real version number. 
Not too worried about tarballs though

**Comment by XECDesign** [comment]
You could fallback to `dpkg-parsechangelog -SVersion` if `git rev-parse` fails, which would give a string like `13.3-1`.

**Comment by timg236** [comment]
Are the APT binaries built from the git repo? I think most people either built from git (non RPiOS) or use the APT binary.

I've added a simple version.
dpkg won't work on other distros but I guess we can try git, then dpkg, then plain version

**Comment by peterharperuk** [comment]
Looks ok

**Comment by XECDesign** [comment]
`Are the APT binaries built from the git repo?`
They are, but only by change.

Normally I'd build just the source package and then kick it off to the build servers to build the .debs for multiple architectures. But the debs for this particular package are built on one machine, from git (at least for now).

I can add some kind of patch to work around the problem on the packaging side for Raspberry Pi OS, but I think other distro maintainers will flag it up too. We've had this problem when libcamera used the same approach.

**Comment by timg236** [comment]
Can we capture the version when the source package is generated?

**Comment by XECDesign** [comment]
> Can we capture the version when the source package is generated?

Yup, that would be how I'd work around it - put it in the changelog and patch the Makefile to read it from there instead.

**Comment by timg236** [comment]
Right, I've captured the latest version date from changelog which should be good enough to identify an APT release and made the git version an optional extra. That's useful for direct builds or topic branches. If git isn't available you just get the APT release date.

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,7 @@
+PKG_VER=$(shell grep rpiboot debian/changelog | head -n1 | sed 's/.*(\(.*\)).*/\1/g')
```
Comment (XECDesign) [inline Makefile:1]: Surely not `rpiboot`

Comment (timg236) [reply Makefile:1]: That's the current format of the changelog

Comment (XECDesign) [reply Makefile:1]: Oh sorry, for some reason I thought this was all for rpi-eeprom.

**Pull request merged**